### PR TITLE
Save model info

### LINKED ISSUE #327: add model info to extxyz
at the moment we add arch to various which is great will be useful and informative... and necessary to add also the info about the model. maybe add in info['{arch}_{model}']=model (use an absolute path where a path was given)

### PR DESCRIPTION
**Pull request by ElliottKasoar**
Resolves #327

For `model_path` (or equivalent inputs e,g, `model`), this adds `mlip_model` as an info label when structures are output.

For strings and paths, this is fine. For already-loaded torch models, it's less clear what the best label is, so I'm open to alternatives.

This will also interact with any implementation of #240, but we can deal with that when we come to it.

@alinelena, was the suggested `info['{arch}_{model}'] = model` replacing `info["arch"]` or additional? To me it doesn't make sense to have `"{arch}_model"` as well as `"arch"`, which is how I originally interpreted it, but perhaps it makes sense to combine them into a single info label, which I just realised may be what you meant.

### CODE REVIEW

File: janus_core/helpers/utils.py [hunk 1]
```diff
@@ -257,6 +257,8 @@ def results_to_info(
     if struct.calc and "arch" in struct.calc.parameters:
         arch = struct.calc.parameters["arch"]
         struct.info["arch"] = arch
+    if struct.calc and "mlip_model" in struct.calc.parameters:
```
Comment (ElliottKasoar) [inline janus_core/helpers/utils.py:260]:
```suggestion
    if struct.calc and "model" in struct.calc.parameters:
```

File: janus_core/helpers/utils.py [hunk 2]
```diff
@@ -474,6 +476,8 @@ def output_structs(
         for image in images:
             if image.calc and "arch" in image.calc.parameters:
                 image.info["arch"] = image.calc.parameters["arch"]
+            if image.calc and "mlip_model" in image.calc.parameters:
```
Comment (ElliottKasoar) [inline janus_core/helpers/utils.py:479]:
```suggestion
            if image.calc and "model" in image.calc.parameters:
```

**Comment by alinelena** [comment]
we need to think about this one, what happens if we have two calculations, both arch=mace but model=small and model=large, how will the extxyz look and similarly if we have some models passed by paths(do we get only the basename for mlip?). I think will be nice to document all this structure of what we want and then review the code accordingly

**Comment by ElliottKasoar** [comment]
> we need to think about this one, what happens if we have two calculations, both arch=mace but model=small and model=large, how will the extxyz look and similarly if we have some models passed by paths(do we get only the basename for mlip?). I think will be nice to document all this structure of what we want and then review the code accordingly

Good question. At the moment I think having the same `arch` is relatively clean, if not perfect, since `info["arch]"`, `info["mlip_model"]`, and labelled results should all just be replaced. This means we can't save both sets of results, but at least I don't think it should be a confusing mix of both.

E.g. `arch=mace, mlip_model=small, mace_energy=x` -> `arch=mace, mlip_model=large, mace_energy=y`

When we change architecture, this is a bit more complicated, since we'd change `info["arch]"`, `info["mlip_model"]`, but keep both sets of results.

E.g. `arch=mace_off, mlip_model=small, mace_off_energy=x` -> `arch=mace_mp, mlip_model=large, mace_off_energy=x, mace_mp_energy=y`.

I implemented it this way for now since it means `mlip_model` essentially works the same as `arch` already does - we can only keep the last one, even if we add new results.

We could change both to potentially be lists (if more than one value), either as two individual lists or a combined entry `{arch}_model={model}`.

This does make things more complicated though, since we'd then want to do something like convert repeated `{arch}_{property}` entries e.g. `mace_energy` into a list too, but only if the model has changed...

> similarly if we have some models passed by paths(do we get only the basename for mlip?)

Regarding paths, at the moment model paths are saved as `str(Path(model_path).expanduser())` e.g. `~` expands to `mlip_model=/home/ek/.cache/mace/46jrkm3v`, but you might get `mlip_model=46jrkm3v` for a relative model path. Maybe we should change this to absolute.

**Comment by alinelena** [comment]
ok I think we need to document very well the behaviour of this with examples since there is no obvious solution.

File: docs/source/user_guide/python.rst [hunk 3]
```diff
@@ -9,3 +9,51 @@ Jupyter Notebook tutorials illustrating the use of currently available calculati
 - `Molecular Dynamics <https://colab.research.google.com/github/stfc/janus-tutorials/blob/main/md.ipynb>`_
 - `Equation of State <https://colab.research.google.com/github/stfc/janus-tutorials/blob/main/eos.ipynb>`_
 - `Phonons <https://colab.research.google.com/github/stfc/janus-tutorials/blob/main/phonons.ipynb>`_
+
+
+Calculation outputs
+===================
+
+By default, calculations performed will modify the underlying `ase.Atoms <https://wiki.fysik.dtu.dk/ase/ase/atoms.html>`_ object
```
Comment (oerc0122) [inline docs/source/user_guide/python.rst:17]: Might this be better as:
```suggestion
By default, calculations performed will modify the underlying :class:`ase.Atoms` object
```
?

File: tests/test_singlepoint_cli.py [hunk 4]
```diff
@@ -54,16 +54,22 @@ def test_singlepoint():
 
     finally:
         # Check atoms can read read, then delete file
```
Comment (oerc0122) [inline tests/test_singlepoint_cli.py:56]: "read read"?

**Review by alinelena — APPROVED** [review]

**Pull request merged**